docs/guide.md

```diff
@@ -262,6 +262,62 @@ if PROFILING:
 To invoke, make any request to your application and it launch a new window
 printing the HTML result from pyinstrument.
 
+### Profile a web request in Litestar
+
+Minimal application setup allowing request profiling.
+
+Middleware ovverides the response to return profiling report as HTML document.
+
+```python
+from asyncio import sleep
+
+from litestar import Litestar, get
+from litestar.middleware import MiddlewareProtocol
+from litestar.types import ASGIApp, Scope, Receive, Send, Message
+from pyinstrument import Profiler
+
+
+class ProfilingMiddleware(MiddlewareProtocol):
+    def __init__(self, app: ASGIApp) -> None:
+        super().__init__(app)  # type: ignore
+        self.app = app
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        profiler = Profiler(interval=0.001, async_mode="enabled")
+
+        profiler.start()
+
+        async def send_wrapper(message: Message) -> None:
+            if message["type"] == "http.response.body":
+                message["body"] = profiler.output_html().encode()
+
+            elif message["type"] == "http.response.start":
+                profiler.stop()
+                message["headers"] = [
+                    (b"content-type", b"text/html; charset=utf-8"),
+                    (b"content-length", str(len(profiler.output_html())).encode()),
+                ]
+
+            await send(message)
+
+        await self.app(scope, receive, send_wrapper)
```

Comment:
It seems maybe a bit wasteful to render the session twice. Does this work?

```suggestion
        profiler.start()
        profile_html: str | None = None

        async def send_wrapper(message: Message) -> None:
            if message["type"] == "http.response.start":
                profiler.stop()
                nonlocal profile_html
                profile_html = profiler.output_html()
                message["headers"] = [
                    (b"content-type", b"text/html; charset=utf-8"),
                    (b"content-length", str(len(profile_html)).encode()),
                ]
            elif message["type"] == "http.response.body":
                message["body"] = profile_html.encode()

            await send(message)

        await self.app(scope, receive, send_wrapper)
```

Also I flipped the order around in the if statement to match the order it (presumably) executes in.

Reply:
of course!